linera-service/src/node_service.rs

```diff
@@ -613,14 +613,14 @@ where
     async fn publish_data_blob(
         &self,
         chain_id: ChainId,
-        blob_content: BlobContent,
+        bytes: Vec<u8>,
     ) -> Result<CryptoHash, Error> {
-        let hash = CryptoHash::new(&blob_content);
+        let hash = CryptoHash::new(&BlobBytes(bytes.clone()));
```

Comment:
(Ah, I see! Because `[u8]` isn't `BcsHashable`!)

Reply:
Yes! And I can make `Vec<u8>` `BcsHashable`, but then it'll need to implement `HasTypeName` also, and it can only do that if it's a struct or enum apparently 😕

Reply:
But that means we're hashing the type name and the content, doesn't it? I think it would be desirable for a data blob's hash to really be just the SHA3 hash of its contents.

No a blocker for _this_ PR, of course! Let's discuss with the team.

Reply:
So `BlobBytes` has `#[repr(transparent)]`, which AFAIU should mean that it's internal representation is the same as a vector of bytes in this case. We just need it to be a struct because when we call `.write` on it inside of `CryptoHash::new`, that will need a proper `HasTypeName` implementation. Which now that it is a struct, it'll be able to get it, but the hash should still be the same as the hash of the internal bytes

Reply:
I'm not 100% sure though, so I can just implement `Hash` manually for `BlobBytes` just in case 🤔

Reply:
Yes, I think `write` also writes the name: https://github.com/linera-io/linera-protocol/blob/b75b3ed673158689590da06fb7b2a76f4230b817/linera-base/src/crypto.rs#L394